jobs/build.Jenkinsfile

```diff
@@ -243,10 +246,52 @@ lock(resource: "build-${params.STREAM}") {
             }
         }
 
+        // Define closure for forking off multi-arch builds.
+        def fork_mArch_jobs = { arches, commit, version, wait ->
```

Comment:
i'm not as happy about the wait here in the case where `WAIT_FOR_RELEASE_JOB` isn't set. Wish there was a better way to make sure the release job didn't get the lock before the build-arch jobs.

Reply:
Maybe we could do something like:

```
while true:
    lock(resource: 'some_resource', skipIfLocked: true) {
      continue
    }
    break
}
```

Reply:
we could put that behind some kind of `waitForLockToBeTaken()` function so it's not as ugly

Reply:
> Maybe we could do something like:

I'll note this isn't foolproof, but will give it a try and we'll see how it looks.